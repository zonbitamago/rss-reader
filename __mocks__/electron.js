const remote ={
  app : {
    getPath :function(str){return '.'}
  }
}

module.exports = {
  require: jest.genMockFunction(),
  match: jest.genMockFunction(),
  app: jest.genMockFunction(),
  remote: remote,
  dialog: jest.genMockFunction()
};
