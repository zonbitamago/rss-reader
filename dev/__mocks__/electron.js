module.exports = {
  ipcRenderer: {
    send: jest.fn()
  },
  remote: {
    app: {
      getPath: () => {
        return "";
      }
    }
  }
};
