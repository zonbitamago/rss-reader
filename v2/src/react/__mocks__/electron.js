module.exports = {
  ipcRenderer: {
    send: jest.genMockFunction()
  }
};
