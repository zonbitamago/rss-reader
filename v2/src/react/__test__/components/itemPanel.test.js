'use strict';
jest.dontMock('../../components/itemPanel.jsx');
import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import {ipcRenderer} from 'electron';
import {mount} from 'enzyme';
import ItemPanel from '../../components/itemPanel.jsx';

test('snapshot', () => {
  var item = {
    link: "https://xxx/1",
    name: 'name1',
    title: 'title1'
  };

  const tree = renderer.create(<ItemPanel item={item}/>);
  expect(tree).toMatchSnapshot();
});

test('記事リンククリック', () => {
  var item = {
    link: [
      {
        href: "https://xxx/1"
      }
    ],
    name: 'name1',
    title: 'title1'
  };

  const wrapper = mount(<ItemPanel item={item}/>);
  wrapper.find('a').simulate('click');

  expect(ipcRenderer.send).toHaveBeenCalled();
});

describe('記事投稿日時',() => {
  test('分表記');

  test('時間表記');

  test('年月日表記');
});
