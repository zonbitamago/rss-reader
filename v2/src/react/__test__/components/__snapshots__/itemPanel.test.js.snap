// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="ui feed"
>
  <div
    className="event"
  >
    <div
      className="label icon"
    >
      <img
        src="http://www.google.com/s2/favicons?domain=xxx"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="summary"
      >
        name1
        <div
          className="date"
        >
          1 Hour Ago
        </div>
      </div>
      <div
        className="extra"
      >
        <a
          href="javascript:void(0);"
          onClick={[Function]}
        >
          title1
        </a>
      </div>
    </div>
  </div>
  <div
    className="ui divider"
  />
</div>
`;
